op-batcher/batcher/driver.go

```diff
@@ -407,6 +408,34 @@ func (l *BatchSubmitter) mainLoop(ctx context.Context, receiptsCh chan txmgr.TxR
 	}
 }
 
+func (l *BatchSubmitter) waitNodeSyncAndClearState() {
+	// Wait for any in flight transactions
+	// to be ingested by the node before
+	// we start loading blocks again.
+	err := l.waitNodeSync()
+	if err != nil {
+		l.Log.Warn("error waiting for node sync", "err", err)
+	}
+	l.clearState(l.shutdownCtx)
+}
+
+// checkExpectedProgress uses the supplied syncStatus to infer
+// whether the node providing the status has made the expected
+// safe head progress given fully submitted channels held in
+// state.
+func (l *BatchSubmitter) checkExpectedProgress(syncStatus eth.SyncStatus) error {
+	verifierCurrentL1 := syncStatus.CurrentL1
+	for _, ch := range l.state.channelQueue {
+		if ch.isFullySubmitted() && // This implies a number of l1 confirmations has passed, depending on how the txmgr was configured
+			!ch.isTimedOut() &&
+			verifierCurrentL1.Number > ch.maxInclusionBlock+l.Config.VerifierL1Confs &&
```

Comment:
Since the sync status tells us where the actual derivation at right now, I think we don't need to add the verifier depth at all.
```suggestion
			verifierCurrentL1.Number > ch.maxInclusionBlock &&
```
This should suffice to tell us that derivation is past the channel inclusion block.

Reply:
I see, I guess I was double counting the verifier confirmation depth in a sense. https://github.com/ethereum-optimism/optimism/blob/d526e425254671aefb2085590b58b15a65789243/op-service/eth/sync_status.go#L5-L18

Reply:
[3df8d16](https://github.com/ethereum-optimism/optimism/pull/12430/commits/3df8d166daaea9e96139c050c390244192766239)